### PR TITLE
Eng 12296 add prerequisite check to plan_upgrade

### PR DESCRIPTION
**Pull request by luyangco**
Now these are following prerequisite checks:
1. Original cluster is licensed for XDCR.
2. Original cluster is running in "xdcr" role.
3. New kit is >=7.2 and is within 1 major release of running version, vice versa for downgrade.
4. Running version is >= 7.2
5. Create warnings if not all the tables are DR tables.

### CODE REVIEW

File: lib/python/voltcli/voltadmin.d/plan_upgrade.py [hunk 1]
```diff
@@ -447,8 +457,8 @@ def createDeploymentForOriginalCluster(runner, xmlString, drSource, origin_clust
 
     dr = et.getroot().find('./dr')
     if dr is None:
-        runner.abort("This cluster doesn't have a DR tag in its deployment file, hence we can't generate online upgrade plan for it. \
-                    Please note add DR tag to your deployment file requires to shutdown and restart the database.")
+        runner.abort("This cluster doesn't have a DR tag in its deployment file, hence we can't generate online upgrade plan for it. " +
```
Comment (xinjiacs) [inline lib/python/voltcli/voltadmin.d/plan_upgrade.py]: "This cluster doesn't have a DR tag in its deployment file, hence we can't generate online upgrade plan."
"In order to add DR tag to the deployment file, users are required to shutdown the database."

File: src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java [hunk 2]
```diff
@@ -25,30 +25,39 @@
...
 
     public static class PrerequisitesCheckNT extends VoltNTSystemProcedure {
-        private final static int MINIMUM_MAJOR_VERSION = 7;
-        private final static int MINIMUM_MINOR_VERSION = 2;
 
         public VoltTable run(String newKitPath, String newRootPath) throws InterruptedException, ExecutionException {
             String ret = checkVoltDBKitExistence(newKitPath);
             String ret2 = checkVoltDBRootExistence(newRootPath);
+            String ret3 = validateXDCRRequirement();
+            String warning = nonFatalCheck();
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java]: Maybe rename "nonFatalCheck" to "checkWarnings"? And add comments to the function what will be checked?

File: src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java [hunk 3]
```diff
@@ -89,25 +109,86 @@ private static String checkVoltDBRootExistence(String newRootPath) {
...
+            }
+
+            return SUCCESS;
+        }
+
+        private static String nonFatalCheck() {
+            StringBuilder warning = new StringBuilder();
+            CatalogContext context = VoltDB.instance().getCatalogContext();
+            for (Table tb : context.database.getTables()) {
+                if (!tb.getTypeName().startsWith("VOLTDB_AUTOGEN_") && !tb.getIsdred()) {
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java]: Can you add a comment why rules out of table names that are prefix with "VOLTDB_AUTOGEN_"?

File: src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java [hunk 4]
```diff
@@ -89,25 +109,86 @@ private static String checkVoltDBRootExistence(String newRootPath) {
...
+                    warning.append(tb.getTypeName()).append(" is not a DR table.").append("\n");
+                }
+            }
+            if (context.getDeployment().getDr() != null && context.getDeployment().getDr().isListen() == false) {
+                warning.append("Target VoltDB cluster is not listening on DR port.(set listen=\"true\" under DR tag of the deployment file)");
+            }
+            if (warning.length() != 0) {
+                return warning.substring(0, warning.length() - 1); // get rid of the '\n'
```
Comment (xinjiacs) [inline src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java]: This logics of removing the '\n' only works for ddl that does not generate "not listening on DR port" warning.

Comment (luyangco) [inline src/frontend/org/voltdb/sysprocs/CheckUpgradePlanNT.java]: Good catch.

**Pull request merged**